app/controllers/collections_controller.rb

```diff
@@ -13,6 +13,24 @@ def form_class
     MyCollectionEditForm
   end
 
+  def show
+    super
+    max_dataset_for_export = 3 * 1024 * 1024 * 1024
+    @dataset_size = 0
+    @can_export = true
+    if @collection.members.respond_to?('each')
+      @collection.members.each do |generic_file|
+        if generic_file.file_size.blank?
+          @can_export = false
+          break
+        end
+        file_size = generic_file.file_size.first.to_i
+        @dataset_size += file_size
+      end
+    end
+    @can_export = (@can_export && @dataset_size < max_dataset_for_export)
```

Comment:
Is 3GB file downloadable? Should it be 
```
@can_export && @dataset_size <= max_dataset_for_export
```

Reply:
3gb has worked in testing. Files that are much larger cause problems.